framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4613,442 @@ private void makeGroundTargetType(
     addDefaultAnnotations(functionalType);
   }
 
+  /**
+   * Return true if {@code type} should be captured.
+   *
+   * <ul>
+   *   <li>{@code type} and {@code typeMirror} must both be declared types.
+   *   <li>{@code type} must not be a raw type, must not have an uninferred type argument, and must
+   *       have a wildcard as a type argument.
+   *   <li>{@code typeMirror} must have a captured type as a type argument.
+   * </ul>
+   *
+   * @param type annotated type to capture
+   * @param typeMirror underlying Java type
+   * @return true if {@code type} should be captured
+   */
+  private boolean shouldCapture(AnnotatedTypeMirror type, TypeMirror typeMirror) {
+    if (type.getKind() != TypeKind.DECLARED) {
+      return false;
+    }
+    if (typeMirror.getKind() != TypeKind.DECLARED) {
+      throw new BugInCF("Expected declared type mirror: %s, type: %s", typeMirror, type);
+    }
+
+    DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
```

Comment:
The name `capturedTypeMirror` was unclear to me when first reading this.  On subsequent readings and after examining call sites, it makes more sense, but could you add documentation and/or renamings to clarify?